public/app/features/dashboard-scene/pages/DashboardScenePage.tsx

```diff
@@ -32,7 +31,9 @@ export function DashboardScenePage({ route, queryParams, location }: Props) {
   // After scene migration is complete and we get rid of old dashboard we should refactor dashboardWatcher so this route reload is not need
   const routeReloadCounter = (location.state as any)?.routeReloadCounter;
 
-  useEffect(() => {
+  // using useLayoutEffect because data links without a slug (referencing same dashboard)
+  //  if data link triggers a query it is lost due to immediate refresh due to url correction
+  useLayoutEffect(() => {
```

Comment:
I'm a bit worried about the performance impact of using this at the DashboardScenePage level. Think we should investigate using some large/complicated dashboards before we commit to this solution.